Error: ‘class QImage’ has no member named ‘flipped’

Almost absolutely user error on my end and I apologize for wasting folks time but I'd appreciate any help I could get. Previously was getting the same error at 33% until I manually installed Clang. Now it reaches 83%. This is being run on a Fedora Distrobox. Attempted both manual & build script.

`[ 83%] Building CXX object CMakeFiles/obsbot-gui.dir/src/gui/PreviewWindow.cpp.o
/home/username/obsbot-camera-control/build/obsbot-camera-control/src/gui/FilterPreviewWidget.cpp: In member function ‘virtual void FilterPreviewWidget::paintGL()’:
/home/username/obsbot-camera-control/build/obsbot-camera-control/src/gui/FilterPreviewWidget.cpp:272:41: error: ‘class QImage’ has no member named ‘flipped’
  272 |         emit processedFrameReady(output.flipped(Qt::Vertical));
      |                                         ^~~~~~~
/home/username/obsbot-camera-control/build/obsbot-camera-control/src/gui/FilterPreviewWidget.cpp: In member function ‘void FilterPreviewWidget::uploadTextureIfNeeded()’:
/home/username/obsbot-camera-control/build/obsbot-camera-control/src/gui/FilterPreviewWidget.cpp:385:37: error: ‘class QImage’ has no member named ‘flipped’
  385 |     QImage glImage = m_currentImage.flipped(Qt::Vertical);
      |                                     ^~~~~~~
[ 87%] Building CXX object CMakeFiles/obsbot-gui.dir/src/common/Config.cpp.o
make[2]: *** [CMakeFiles/obsbot-gui.dir/build.make:189: CMakeFiles/obsbot-gui.dir/src/gui/FilterPreviewWidget.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:90: CMakeFiles/obsbot-gui.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
`

--- a/src/gui/FilterPreviewWidget.cpp
+++ b/src/gui/FilterPreviewWidget.cpp
@@ -5,9 +5,20 @@
 #include <QVector2D>
 #include <QVideoFrame>
 #include <QtMath>
+#include <QtGlobal>
 #include <cmath>
 
 namespace {
+
+// Qt compatibility: flipped() added in Qt 5.10, mirrored() deprecated in Qt 6
+inline QImage flipVertical(const QImage &image)
+{
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
+    return image.flipped(Qt::Vertical);
+#else
+    return image.mirrored(false, true);
+#endif
+}
 
 const char *kVertexShaderSource = R"(#version 330 core
 layout(location = 0) in vec2 a_position;
@@ -269,7 +280,7 @@ void FilterPreviewWidget::paintGL()
                      GL_RGBA, GL_UNSIGNED_BYTE, output.bits());
         m_framebuffer->release();
 
-        emit processedFrameReady(output.flipped(Qt::Vertical));
+        emit processedFrameReady(flipVertical(output));
         m_emitPending = false;
     }
 
@@ -382,7 +393,7 @@ void FilterPreviewWidget::uploadTextureIfNeeded()
         m_texture->bind();
     }
 
-    QImage glImage = m_currentImage.flipped(Qt::Vertical);
+    QImage glImage = flipVertical(m_currentImage);
     glPixelStorei(GL_UNPACK_ALIGNMENT, 1);
     glTexSubImage2D(GL_TEXTURE_2D, 0, 0, 0,
                     frameSize.width(), frameSize.height(),

--- a/src/gui/TrackingControlWidget.cpp
+++ b/src/gui/TrackingControlWidget.cpp
@@ -258,7 +258,11 @@ void TrackingControlWidget::updateFromState(const CameraController::CameraState 
     // 2. Not during user action AND
     // 3. Command completion timer has expired AND
     // 4. Camera is not settling
-    bool shouldBeChecked = (state.aiMode != Device::AiWorkModeNone);
+    // For Tiny2: aiMode determines tracking state
+    // For non-Tiny2 (original Tiny): use autoFramingEnabled flag
+    bool shouldBeChecked = m_tiny2Capabilities
+        ? (state.aiMode != Device::AiWorkModeNone)
+        : state.autoFramingEnabled;
     bool commandInFlight = m_commandTimer->isActive();
     bool isSettling = m_controller->isSettling();
 

